vcs-diff-lint-csdiff-ruff

```diff
@@ -24,29 +24,33 @@ def ruff_check():
     return json.loads(out)
 
 
-def ruff_code_to_name(code):
+def ruff_code_to_name():
     """
-    Convert noqa code e.g. F401 to its human-readable name, e.g. unused-import
+    Introspect ruff and map all possible noqa codes to a human-readable names.
     """
-    # This implementation will likely kill all ruff performance benefits
-    cmd = ["ruff", "rule", code, "--output-format=json"]
+    cmd = ["ruff", "rule", "--all", "--output-format=json"]
     with Popen(cmd, stdout=PIPE) as proc:
         out, _err = proc.communicate(timeout=60)
-    data = json.loads(out)
-    return data["name"]
+    the_map = {}
+    array = json.loads(out)
+    for rule in array:
+        the_map[rule["code"]] = rule['name']
+    return the_map
```

Comment:
Nitpick but so many lines can be saved by comprehension here

```python
return {x["code"]: x["name"] for x in array}
```

Reply:
Indeed, updated :)